crates/accelerate/src/elide_permutations.rs

```diff
@@ -0,0 +1,104 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2024
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use numpy::PyReadonlyArray1;
+use pyo3::prelude::*;
+
+use qiskit_circuit::dag_circuit::{DAGCircuit, NodeType};
+use qiskit_circuit::operations::{Operation, Param};
+use qiskit_circuit::Qubit;
+
+/// Run the ElidePermutations pass on `dag`.
+/// Args:
+///     dag (DAGCircuit): the DAG to be optimized.
+/// Returns:
+///     An `Option`: the value of `None` indicates that no optimization was
+///     performed and the original `dag` should be used, otherwise it's a
+///     tuple consisting of the optimized DAG and the induced qubit permutation.
+#[pyfunction]
+fn run(py: Python, dag: &mut DAGCircuit) -> PyResult<Option<(DAGCircuit, Vec<usize>)>> {
+    let permutation_gate_names = ["swap".to_string(), "permutation".to_string()];
+    let op_counts = dag.count_ops(py, false)?;
+    if !permutation_gate_names
+        .iter()
+        .any(|name| op_counts.contains_key(name))
+    {
+        return Ok(None);
+    }
+    let mut mapping: Vec<usize> = (0..dag.num_qubits()).collect();
+
+    // note that DAGCircuit::copy_empty_like clones the interners
+    let mut new_dag = dag.copy_empty_like(py, "alike")?;
+    for node_index in dag.topological_op_nodes()? {
+        if let NodeType::Operation(inst) = &dag.dag[node_index] {
+            match (inst.op.name(), inst.condition()) {
+                ("swap", None) => {
+                    let qargs = dag.get_qargs(inst.qubits);
+                    let index0 = qargs[0].0 as usize;
+                    let index1 = qargs[1].0 as usize;
+                    let prev0 = mapping[index0];
+                    let prev1 = mapping[index1];
+                    mapping[index0] = prev1;
+                    mapping[index1] = prev0;
```

Comment:
You can use https://doc.rust-lang.org/std/vec/struct.Vec.html#method.swap here:

```suggestion
                    mapping.swap(index0, index1);
```

Reply:
I knew this existed! Thanks!

Reply:
Done!